pkg/cli/idle/idle.go

```diff
@@ -268,6 +268,10 @@ func (o *IdleOptions) calculateIdlableAnnotationsByService(infoVisitor func(reso
 		}
 		// just get the unversioned version of this
 		gk := schema.GroupKind{Group: gv.Group, Kind: ref.Kind}
+		if gk.Kind == "StatefulSet" {
+			fmt.Println("Idling StatefulSet is not supported yet.")
+			return nil, nil
```

Comment:
Is there a reason you need to print here instead of returning an error?
```suggestion
			return nil, fmt.Errorf("idling StatefulSets is not supported")
```

Reply:
Also, don't you need three return values here?  
```suggestion
			return nil, nil, fmt.Errorf("idling StatefulSets is not supported")
```

Reply:
Thanks @Miciah ! return  is inside 	getController := func(ref namespacedOwnerReference) (metav1.Object, error) {

so I will add return nil, fmt.Errorf("idling StatefulSets is not supported") as can't return 3 values.

Reply:
> Is there a reason you need to print here instead of returning an error?

It works fine however the right approach will be to add an error in the return as you mentioned as the return type of the function is error.

Reply:
> return is inside getController := func(ref namespacedOwnerReference) (metav1.Object, error) {

Ah, I see.  GitHub doesn't show the indentation correctly.

Reply:
> Thanks @Miciah ! return is inside getController := func(ref namespacedOwnerReference) (metav1.Object, error) {
> 
> so I will add return nil, fmt.Errorf("idling StatefulSets is not supported") as can't return 3 values.



@Miciah after adding this I don't see the error message like I do see with this commit as follows -

[miheer@localhost oc]$ ./oc idle nginx
WARNING: idling when network policies are in place may cause connections to bypass network policy entirely
Idling StatefulSet is not supported yet.
error: unable to mark the service "nginx/nginx" as idled.
Make sure that the service is not already marked as idled and that it is associated with resources that can be scaled.
See 'oc idle -h' for help and examples.
[miheer@localhost oc]$ 

After adding the return  nil, fmt.Errorf("idling StatefulSets is not supported") it does not show the error

[miheer@localhost oc]$ ./oc idle nginx
WARNING: idling when network policies are in place may cause connections to bypass network policy entirely
error: unable to mark the service "nginx/nginx" as idled.
Make sure that the service is not already marked as idled and that it is associated with resources that can be scaled.
See 'oc idle -h' for help and examples.

Reply:
BTW I had to make the code changes where get controller is called and where it's error is captured if you want to return error.

Reply:
[miheer@localhost oc]$ ./oc idle nginx
WARNING: idling when network policies are in place may cause connections to bypass network policy entirely
error: no valid scalable resources found to idle: unable to calculate scalable resources for service nginx/nginx: Idling StatefulSet is not supported yet.
[miheer@localhost oc]$